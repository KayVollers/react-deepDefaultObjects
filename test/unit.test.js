import React from 'react';
import StatelessComp from './statelessComp';
import StatefulComp from './statefulComp';
import renderer from 'react-test-renderer';

const props = {
    originMessage:'unchanged',
    originNested: {
        originMessage: 'unchanged',
    },
    nestedMessages: {
        nestMessage: 'unchanged',
        originNestMessage: 'unchanged',
        originNestArray: ['unchanged'],
    },
};

test('stateless react component', () => {
    const tree = renderer.create(
        <StatelessComp
            {...props}
        />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('stateful react component', () => {
    const tree = renderer.create(
        <StatefulComp
            {...props}
        />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});