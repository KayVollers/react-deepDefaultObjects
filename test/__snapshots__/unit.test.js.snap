// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stateful react component 1`] = `
<div>
  Stateful 
  <p />
   
  {
	"originMessage": "unchanged",
	"originNested": {
		"originMessage": "unchanged"
	},
	"nestedMessages": {
		"nestMessage": "unchanged",
		"originNestMessage": "unchanged",
		"originNestArray": [
			"unchanged"
		],
		"defaultNestMessage": "default"
	},
	"defaultMessage": "default"
}
</div>
`;

exports[`stateless react component 1`] = `
<div>
  Stateless 
  <p />
   
  {
	"originMessage": "unchanged",
	"originNested": {
		"originMessage": "unchanged"
	},
	"nestedMessages": {
		"nestMessage": "unchanged",
		"originNestMessage": "unchanged",
		"originNestArray": [
			"unchanged"
		],
		"defaultNestMessage": "default"
	},
	"defaultMessage": "default"
}
</div>
`;
